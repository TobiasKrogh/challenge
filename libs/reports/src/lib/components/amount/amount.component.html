<ng-container *ngIf="amount">
  <ng-container *ngIf="suffix">{{ suffix }} </ng-container>
  {{ amount | formatAmount }}
  {{ 'CURRENCY' | translate }}
</ng-container>
