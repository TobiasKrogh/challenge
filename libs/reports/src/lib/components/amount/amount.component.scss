:host {
  display: block;
  font: var(--font-data--emphasizes);
}
