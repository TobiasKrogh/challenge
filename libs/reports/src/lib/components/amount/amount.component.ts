import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

@Component({
  selector: 'reports-amount',
  templateUrl: './amount.component.html',
  styleUrls: ['./amount.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AmountComponent {
  @Input() amount: number | undefined;
  @Input() suffix: string | undefined;
}
