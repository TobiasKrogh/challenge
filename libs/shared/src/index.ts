export * from './lib/components/message/message.component';
export * from './lib/functions/normalize-date';
export * from './lib/pipes/format-amount.pipe';
export * from './lib/pipes/format-date.pipe';
