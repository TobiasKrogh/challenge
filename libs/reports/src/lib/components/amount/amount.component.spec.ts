import { ComponentFixture, TestBed } from '@angular/core/testing';
import { TranslatePipe } from '@ngx-translate/core';
import { FormatAmountPipe } from '@challenge/shared';
import { MockPipes } from 'ng-mocks';

import { AmountComponent } from './amount.component';

describe('AmountComponent', () => {
  let component: AmountComponent;
  let fixture: ComponentFixture<AmountComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        MockPipes(TranslatePipe),
        FormatAmountPipe,
        AmountComponent,
      ],
    }).compileComponents();

    fixture = TestBed.createComponent(AmountComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    fixture.detectChanges();
    expect(component).toBeTruthy();
  });

  describe('When rendering', () => {
    it('Then it matches snapshot (no amount given)', () => {
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });

    it('Then it matches snapshot (amount provided)', () => {
      component.amount = 50;
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });

    it('Then it matches snapshot (suffix and amount provided)', () => {
      component.amount = 50;
      component.suffix = 'suffix';
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });
  });
});
