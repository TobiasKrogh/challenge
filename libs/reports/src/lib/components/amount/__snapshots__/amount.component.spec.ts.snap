// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountComponent When rendering Then it matches snapshot (amount provided) 1`] = `
DebugElement {
  "nativeNode": <div
    id="root2"
  >
    
     50.00 

    
    
  </div>,
}
`;

exports[`AmountComponent When rendering Then it matches snapshot (no amount given) 1`] = `
DebugElement {
  "nativeNode": <div
    id="root1"
  >
    
  </div>,
}
`;

exports[`AmountComponent When rendering Then it matches snapshot (suffix and amount provided) 1`] = `
DebugElement {
  "nativeNode": <div
    id="root3"
  >
    suffix 
    
    
     50.00 

    
    
  </div>,
}
`;
